service/history/nDCTransactionMgr.go

```diff
@@ -295,3 +318,54 @@ func (r *nDCTransactionMgrImpl) loadNDCWorkflow(
 	}
 	return newNDCWorkflow(ctx, r.domainCache, r.clusterMetadata, context, msBuilder, release), nil
 }
+
+func (r *nDCTransactionMgrImpl) reapplyEvents(
+	ctx ctx.Context,
+	reapplyEvents *persistence.WorkflowEvents,
+) error {
+
+	hasReappliableEvent := false
```

Comment:
since you are already doing pre processing, i.e. checking if there are signals to be reapplied, 
might as well filter out the non signal events

Reply:
done